strawberry_django/mutations/resolvers.py

```diff
@@ -185,6 +159,87 @@ def parse_input(info: Info, data: Any):
     return data
 
 
+def prepare_create_update(
+    *,
+    info: Info,
+    instance: Model,
+    data: dict[str, Any],
+    full_clean: bool | FullCleanOptions = True,
+) -> tuple[Model, dict, list, list]:
+    """Prepare data for updates and creates.
+
+    This method is a helper function for the create and
+    update resolver methods.  It's to prepare the data
+    for updating or creating.
+    """
+    model = instance.__class__
+    fields = get_model_fields(model)
+    files: list[
+        tuple[
+            models.FileField,
+            Upload | Literal[False],
+        ]
+    ] = []
+    m2m: list[tuple[ManyToManyField | ForeignObjectRel, Any]] = []
+    direct_field_values = {}
+
+    if dataclasses.is_dataclass(data):
+        data = vars(data)
+
+    for name, value in data.items():
+        field = fields.get(name)
+        direct_field_value = True
+
+        if field is None or value is UNSET:
+            # Dont use these, fallback to model defaults.
+            direct_field_value = False
+            continue
+
+        if isinstance(field, models.FileField):
+            if value is None:
+                # We want to reset the file field value when None was passed in the
+                # input, but `FileField.save_form_data` ignores None values. In that
+                # case we manually pass False which clears the file.
+                value = False  # noqa: PLW2901
+
+            # set FileFields at the same time so their hooks can use other set values
+            files.append((field, value))
+            direct_field_value = False
+            continue
+
+        if isinstance(field, (ManyToManyField, ForeignObjectRel)):
+            # m2m will be processed later
+            m2m.append((field, value))
+            direct_field_value = False
+            continue
+
+        if isinstance(field, models.ForeignKey) and isinstance(
+            value,
+            # We are using str here because strawberry.ID can't be used for isinstance
+            (ParsedObject, str),
+        ):
+            value, value_data = _parse_data(  # noqa: PLW2901
+                info,
+                field.related_model,
+                value,
+            )
+            if value is None and not value_data:
+                value = None  # noqa: PLW2901
+            elif value is None:
+                value = field.related_model._default_manager.create(  # noqa: PLW2901
+                    **value_data,
+                )
+            else:
+                update(info, value, value_data, full_clean=full_clean)
+
+        if direct_field_value:
```

Comment:
From what I can tell this if will always be `True`, because it gets set to True at the beggining of the for loop, and everytime it gets set to `False` you call `continue`

Reply:
Honestly, the original had continue statements without comments.  Personally didn't see the point - but decided to leave them.  There were no comments why they are called.  But yes, happy to remove them, they don't seem to serve any purpose. Unless I'm missing what continue does.

Reply:
I've reworked it a little. Both versions seem to work fine though.